src/Symfony/Component/OptionsResolver/OptionConfigurator.php

```diff
@@ -0,0 +1,187 @@
+        $this->resolver->setDefined($name);
+    }
+
+    /**
+     * Adds allowed values for this option.
+     *
+     * @param mixed ...$allowedValues One or more acceptable values/closures
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function addAllowedValues(...$allowedValues): self
+    {
+        $this->resolver->addAllowedValues($this->name, $allowedValues);
+
+        return $this;
+    }
+
+    /**
+     * Sets allowed types for this option.
+     *
+     * @param string ...$allowedTypes One or more accepted types
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function addAllowedTypes(string ...$allowedTypes): self
+    {
+        $this->resolver->addAllowedTypes($this->name, $allowedTypes);
+
+        return $this;
+    }
+
+    /**
+     * Adds a normalizer for this option.
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function addNormalizer(\Closure $normalizer): self
+    {
+        $this->resolver->addNormalizer($this->name, $normalizer);
+
+        return $this;
+    }
+
+    /**
+     * Prepends a normalizer for this option.
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function prependNormalizer(\Closure $normalizer): self
+    {
+        $this->resolver->addNormalizer($this->name, $normalizer, true);
+
+        return $this;
+    }
+
+    /**
+     * Adds allowed types for this option.
+     *
+     * @param string ...$types One or more accepted types
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function allowedTypes(string ...$types): self
+    {
+        $this->resolver->setAllowedTypes($this->name, $types);
+
+        return $this;
+    }
+
+    /**
+     * Sets allowed values for this option.
+     *
+     * @param mixed ...$values One or more acceptable values/closures
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function allowedValues(...$values): self
+    {
+        $this->resolver->setAllowedValues($this->name, $values);
+
+        return $this;
+    }
+
+    /**
+     * Sets the default value for this option.
+     *
+     * @param mixed $value The default value of the option
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function default($value): self
+    {
+        $this->resolver->setDefault($this->name, $value);
+
+        return $this;
+    }
+
+    /**
+     * Defines an option configurator with the given name.
+     */
+    public function define(string $option): self
```

Comment:
IMO we don't need this method. The problem is it causes indention problems like we have with the Config Builder. See your doc that shows this: https://github.com/symfony/symfony-docs/pull/12426/files#diff-61b5eb7a03a5b6f645182e6c37bd7ee2R806

Reply:
It's up to you to do it this way, it's a matter of preference, we're not forcing anyone to use it, while those who like it will be able to. It's no big deal imo.

Reply:
also note that we already did it for other configurators:
https://symfony.com/blog/new-in-symfony-3-4-php-based-configuration-for-services-and-routes